tests/src/main/java/com/hazelcast/simulator/tests/topic/ReliableTopicTest.java

```diff
@@ -0,0 +1,168 @@
+    public int threadCount = 3;
+    public int listenersPerTopic = 2;
+    public String basename = "reliableTopic";
+
+
+    private IAtomicLong totalExpectedCounter;
+    private ITopic[] topics;
+    private TestContext testContext;
+    private HazelcastInstance hz;
+    private List<StressMessageListener> listeners;
+
+    @Setup
+    public void setup(TestContext testContext) throws Exception {
+        this.testContext = testContext;
+        hz = testContext.getTargetInstance();
+
+        totalExpectedCounter = hz.getAtomicLong(testContext.getTestId() + ":TotalExpectedCounter");
+        topics = new ITopic[topicCount];
+        listeners = new LinkedList<StressMessageListener>();
+        int listenerIdCounter = 0;
+        for (int k = 0; k < topics.length; k++) {
+            ITopic<Long> topic = hz.getReliableTopic(basename + "-" + k);
+            topics[k] = topic;
+            for (int l = 0; l < listenersPerTopic; l++) {
+                StressMessageListener topicListener = new StressMessageListener(listenerIdCounter);
+                listenerIdCounter++;
+                topic.addMessageListener(topicListener);
+                listeners.add(topicListener);
+            }
+        }
+    }
+
+    @Run
+    public void run() {
+        ThreadSpawner spawner = new ThreadSpawner(testContext.getTestId());
+        for (int k = 0; k < threadCount; k++) {
+            spawner.spawn(new Worker());
+        }
+        spawner.awaitCompletion();
+    }
+
+    private class Worker implements Runnable {
+        private final Random random = new Random();
+
+        @Override
+        public void run() {
+            long count = 0;
+            while (!testContext.isStopped()) {
+
+                ITopic topic = getRandomTopic();
+                long msg = nextMessage();
+                count += msg;
+                topic.publish(msg);
+            }
+            totalExpectedCounter.addAndGet(count);
+        }
+
+        private ITopic getRandomTopic() {
+            int index = random.nextInt(topics.length);
+            return topics[index];
+        }
+
+        private long nextMessage() {
+            long msg = random.nextLong() % 1000;
+            if (msg < 0) {
+                msg = -msg;
+            }
+            return msg;
+        }
+    }
+
+    @Verify(global = true)
+    public void verify() {
+
+        final long expectedCount = totalExpectedCounter.get();
+
+        assertTrueEventually(new AssertTask() {
+            @Override
+            public void run() throws Exception {
+                long actualCount = 0;
+                for (StressMessageListener topicListener : listeners) {
+                    actualCount += topicListener.received;
+                }
+                assertEquals("published messages don't match received messages", expectedCount, actualCount);
+            }
+        });
+    }
+
+    public class StressMessageListener implements ReliableMessageListener<Long> {
```

Comment:
You don't need to use this interface btw. You can also use the regular MessageListener.